components/script/script_runtime.rs

```diff
@@ -826,7 +832,46 @@ impl StreamConsumer {
 }
 
 #[allow(unsafe_code)]
-unsafe extern "C" fn report_stream_error_callback(_cx: *mut JSContext, error_code: usize) {
+unsafe extern "C" fn ConsumeStreamCallback(
+    _cx: *mut RawJSContext,
+    obj: HandleObject,
+    _mimeType: MimeType,
+    _consumer: *mut JSStreamConsumer,
+) {
+    let cx = JSContext::from_ptr(_cx);
+    let global = GlobalScope::from_context(*cx);
+    let returnValue = Promise::new(&global);
+    let mut handleobj = UndefinedValue();
+    GetProxyPrivate(*obj, &mut handleobj);
+    rooted!(in(*cx) let target = handleobj.to_object());
+    if let Ok(unwrappedSource) = root_from_handleobject::<Response>(target.handle(), *cx) {
+        let mimetype = unwrappedSource.Headers().extract_mime_type();
+        if std::str::from_utf8(&mimetype).unwrap() != "application/wasm" {
+            return returnValue.reject_error(Error::Type(
+                "Response has unsupported MIME type".to_string(),
+            ));
+        }
+        if unwrappedSource.Type().as_str() != "Basic" ||
+            unwrappedSource.Type().as_str() != "Cors" ||
+            unwrappedSource.Type().as_str() != "Default"
+        {
+            return returnValue.reject_error(Error::Type(
+                "Response.type must be 'basic', 'cors' or 'default'".to_string(),
+            ));
+        }
+        if !unwrappedSource.Ok() {
```

Comment:
Let's add some comments like `// step 2.5` to make it more clear which parts of the spec are being implemented by the code.

Reply:
Done